src/Umbraco.Forms.Integrations.Automation.Zapier/Controllers/PollingController.cs

```diff
@@ -0,0 +1,114 @@
+using Umbraco.Cms.Core.Services;
+#else
+using System.Configuration;
+
+using Umbraco.Web.WebApi;
+using Umbraco.Core.Services;
+#endif
+
+namespace Umbraco.Forms.Integrations.Automation.Zapier.Controllers
+{
+    public class PollingController : UmbracoApiController
+    {
+        private readonly ZapierSettings Options;
+
+        private readonly IFormService _formService;
+
+        private readonly IWorkflowService _workflowService;
+
+        private readonly IRecordReaderService _recordReaderService;
+
+        private readonly IUserValidationService _userValidationService;
+
+#if NETCOREAPP
+        public PollingController(IOptions<ZapierSettings> options, IFormService formService, IWorkflowService workflowService,
+            IRecordReaderService recordReaderService, IUserValidationService userValidationService)
+#else
+        public PollingController(IFormService formService, IWorkflowService workflowService, 
+            IRecordReaderService recordReaderService, IUserValidationService userValidationService)
+#endif
+        {
+#if NETCOREAPP
+            Options = options.Value;
+#else
+            Options = new ZapierSettings(ConfigurationManager.AppSettings);
+#endif
+
+            _formService = formService;
+
+            _workflowService = workflowService;
+
+            _recordReaderService = recordReaderService;
+
+            _userValidationService = userValidationService;
+        }
+
+        public List<Dictionary<string, string>> GetFormsData()
+        {
+            var formsData = new List<Dictionary<string, string>>();
+
+            string username = string.Empty;
+            string password = string.Empty;
+
+#if NETCOREAPP
+            if (Request.Headers.TryGetValue(Constants.ZapierAppConfiguration.UsernameHeaderKey,
+                    out var usernameValues))
+                username = usernameValues.First();
+            if (Request.Headers.TryGetValue(Constants.ZapierAppConfiguration.PasswordHeaderKey,
+                    out var passwordValues))
+                password = passwordValues.First();
+#else
+                        if (Request.Headers.TryGetValues(Constants.ZapierAppConfiguration.UsernameHeaderKey,
+                                out var usernameValues))
+                            username = usernameValues.First();
+                        if (Request.Headers.TryGetValues(Constants.ZapierAppConfiguration.PasswordHeaderKey,
+                                out var passwordValues))
+                            password = passwordValues.First();
+#endif
+
+            if (string.IsNullOrEmpty(username) || string.IsNullOrEmpty(password)) return null;
+
+            var isAuthorized = _userValidationService.Validate(username, password, Options.UserGroup).GetAwaiter().GetResult();
+            if (!isAuthorized) return null;
+
+            var zapierWorkflowId = new Guid("d05b95e5-86f8-4c31-99b8-4ec7fc62a787");
+
+            // 1. get forms
+            var forms = _formService.Get();
```

Comment:
Yes, that's right - but that's only because in V9 we did away with storing forms on disk and supporting only "forms in the database".  Hence `IFormStorage` was no longer needed, and removed.

It is still needed in V8 though when form definitions are stored in files on disk, and won't be removed in any subsequent V8 release, so it's safe, and necessary, to use.

Reply:
Updated.